Network/XmlRpc/Internals.hs

```diff
@@ -482,6 +562,10 @@ readBool s = errorRead readsBool "Error parsing boolean" s
           readsBool "0" = [(False,"")]
           readsBool _ = []
 
+readNil :: Monad m => (Maybe XR.Value) -> Err m (Maybe Value)
+readNil Nothing = pure Nothing
```

Comment:
Please use `return` instead of `pure`, that way it will continue to compile under pre-AMP versions of GHC/base (e.g. GHC 7.8).